chat.go

```diff
@@ -80,6 +86,7 @@ func GetChatRoom(name string, password string, joinable, fixed bool) (room *Chat
 	return room
 }
 
+// run is the main loop that runs and handles events for a ChatRoom.
 func (cr *ChatRoom) run() {
 	defer func() {
```

Comment:
Yeah, they should probably be private. It was more of a self-reminder which ones are helper functions and which ones should actually be called.

I will put this in the close func then.